README.md

```diff
@@ -1,223 +1,350 @@
-
-This workspace is divided in 6 macro-areas, (1) *protocols* several Stratum library, (2) *roles* the
-actual Sv2 roles, (3) *utils*, (4) *examples*, (5) *test* integration tests, (6) *experimental* not yet
-specified part of the protocol or extensions.
-
-Every dependency related to benchmarking and testing can be always opted out so we never list
-them under **External dependencies**.
-
-### PROTOCOLS
-
-Under *protocols* there is *v1* that contain an Sv1 library and *v2* that contain several Sv2
-libraries.
-
-### PROTOCOLS/V1
-TODO
-
-### PROTOCOLS/V2
-TODO
-
-### PROTOCOLS/V2/CONST-SV2
-
-A bunch of Sv2 related constants.
+* These libraries can all be used, or only a subset of them can be used, depending on the
+  functionality desired by the end user. Some examples of individuals who benefit from the
+  libraries in this repository are as follows (note that not all features are completed yet):
+    * A miner who runs a mining farm with Sv2-compatible mining firmware mining to a 
+      Sv2-compatible pool can use this library as a proxy which allows them to use the Group
+      Channels, which reduce their bandwidth consumption and bolster their efficiency. Standard
+      Channels are also supported in here, but are not as efficient as Group Channels.
+    * A miner who runs a mining farm with Sv2-compatible mining firmware mining to a Sv2-compatible
+      pool, who wants to select their own transactions to build their own `blocktemplate`, can use
+      this library as a proxy in conjunction with a Bitcoin Core node (with the Template Provider
+      logic) to do so.
+    * A miner who runs a mining farm with SV1-compatible mining firmware mining to a Sv2-compatible
+      pool, who wants to gain some of the security and efficiency improvements that Sv2 offers over
+      Stratum V1 (Sv1) (by using Extended Channels). This Sv1<->Sv2 miner proxy does not support
+      all the features of Sv2, therefore it should be used as a temporary measure before completely
+      upgrading to Sv2-compatible firmware.
+    * A pool that wants to build a Sv2 compatible pool server, complete with all Sv2 channel
+      support (Standard, Group, and Extended channels).
+    * A pool that wants to begin supporting Sv2 but is not yet ready to completely overhaul their
+      existing Sv1 server can use these libraries to construct a Sv2<->Sv1 proxy server. This proxy
+      server lives on the upstream pool end under the pool's control, between the downstream
+      devices on the miner's side (either a miner's proxy or the mining devices themselves), and
+      the most upstream Sv1 pool server. This most upstream Sv1 pool server is likely the already
+      existing pool server logic. In this way, pools can begin supporting Sv2 without waiting to
+      redo their entire server infrastructure to be Sv2 compatible only. This Sv2<->Sv1 pool proxy
+      will only allow the pool to support Standard channels (not Group or Extended), therefore this
```

Comment:
I don't see why this must be the case.
As a first impression it seems possible to me have something like that:
mining devices <= standard channels => miner's proxy <- group channel -> pool's proxy v2 to v1 <- v1 -> pool

Btw not 100% sure.